src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -1,201 +0,0 @@
-            public readonly Encoding Encoding;
-            public readonly int Length;
-            public readonly bool ContainsDirectives;
-
-            public SyntaxTreeInfo(
-                string filePath,
-                ParseOptions options,
-                ITextAndVersionSource textSource,
-                LoadTextOptions loadTextOptions,
-                Encoding encoding,
-                int length,
-                bool containsDirectives)
-            {
-                FilePath = filePath ?? string.Empty;
-                Options = options;
-                TextSource = textSource;
-                LoadTextOptions = loadTextOptions;
-                Encoding = encoding;
-                Length = length;
-                ContainsDirectives = containsDirectives;
-            }
-
-            internal bool TryGetText([NotNullWhen(true)] out SourceText? text)
-            {
-                if (TextSource.TryGetValue(LoadTextOptions, out var textAndVersion))
-                {
-                    text = textAndVersion.Text;
-                    return true;
-                }
-
-                text = null;
-                return false;
-            }
-
-            internal SourceText GetText(CancellationToken cancellationToken)
-                => TextSource.GetValue(LoadTextOptions, cancellationToken).Text;
-
-            internal async Task<SourceText> GetTextAsync(CancellationToken cancellationToken)
-            {
-                var textAndVersion = await TextSource.GetValueAsync(LoadTextOptions, cancellationToken).ConfigureAwait(false);
-                return textAndVersion.Text;
-            }
-
-            internal SyntaxTreeInfo WithFilePath(string path)
-            {
-                return new SyntaxTreeInfo(
-                    path,
-                    Options,
-                    TextSource,
-                    LoadTextOptions,
-                    Encoding,
-                    Length,
-                    ContainsDirectives);
-            }
-
-            internal SyntaxTreeInfo WithOptionsAndLengthAndContainsDirectives(ParseOptions options, int length, bool containsDirectives)
-            {
-                return new SyntaxTreeInfo(
-                    FilePath,
-                    options,
-                    TextSource,
-                    LoadTextOptions,
-                    Encoding,
-                    length,
-                    containsDirectives);
-            }
-
-            internal SyntaxTreeInfo WithOptions(ParseOptions options)
-            {
-                return new SyntaxTreeInfo(
-                    FilePath,
-                    options,
-                    TextSource,
-                    LoadTextOptions,
-                    Encoding,
-                    Length,
-                    ContainsDirectives);
-            }
-        }
-
-        internal sealed class RecoverableSyntaxRoot<TRoot> : WeaklyCachedRecoverableValueSource<TRoot>
```

Comment:
without this RecoverableText is the only derivation of WeaklyCachedRecoverableValueSource.  I will strongly consider merging those into one type in a followup PR.